### PR TITLE
Check hostname before redirecting to new URL

### PR DESCRIPTION
**Pull request by bitboxer**
Closes [#120](https://github.com/railslove/recover-backlog/issues/120)

### CODE REVIEW

File: pages/qr.tsx [hunk 1]
```diff
@@ -31,7 +30,12 @@ export default function QRCodePage() {
           undefined,
           videoEl.current
         )
-        window.location.href = appendUrlParams(result.getText())
+        const qrUrl = new URL(result.getText())
+        if (qrUrl.hostname === 'rcvr.app') {
```
Comment (bonflintstone) [inline pages/qr.tsx:34]: I don't think hard coding this here makes a lot of sense, does it?

Comment (bitboxer) [reply pages/qr.tsx:34]: Do we have a config variable for this somewhere that I can use_

Comment (salimhb) [reply pages/qr.tsx:34]: There's an ENV variable. But I don't think there's a need for that. Simply check that it matches `location.hostname`.

Comment (bitboxer) [reply pages/qr.tsx:34]: Ah, right. Perfect 🥳

File: pages/qr.tsx [hunk 2]
```diff
@@ -31,7 +30,12 @@ export default function QRCodePage() {
...
+          window.location.href = appendUrlParams(qrUrl)
+        } else {
+          console.error('Found wrong url host', qrUrl.hostname)
```
Comment (salimhb) [inline pages/qr.tsx:37]: So for a user who doesn't see the console, we're failing silently?

I was against this from [start](https://github.com/railslove/recover-backlog/issues/120#issuecomment-760251270) and was later open to vetting, but failing completely is not ideal. What if I used rcvr.app/qr to scan a code from care.rcvr.app.
I suggest in this case, we simply show a warning message. No need for a fancy UI, just simple `confirm("Warning, this QR code is not part of RecoverApp, you may continue there, but any data you enter will be sent to ${qrUrl.hostname}")` would do.

Comment (bitboxer) [reply pages/qr.tsx:37]: Okay, will do so.

**Review by salimhb — APPROVED** [review]

**Pull request merged**